backend/wmg/api/common/rollup.py

```diff
@@ -85,53 +83,58 @@ def _add_missing_combinations_to_gene_expression_df_for_rollup(
         and likely greater size than the input gene expression dataframe that includes combinations
         for which numeric values should be aggregated during the rollup operation.
     """
-    # extract group-by terms and queried genes from the input dataframes
-    # if a queried gene is not present in the input dot plot dataframe, we can safely
-    # ignore it as it need not be rolled up anyway.
-    group_by_terms = list(universal_set_cell_counts_df.index.names)
-    genes = list(set(gene_expression_df["gene_ontology_term_id"]))
+    if gene_expression_df.shape[0] == 0:
+        return gene_expression_df
 
-    # get the names of the numeric columns
     numeric_columns = list(
         gene_expression_df.columns[[np.issubdtype(dtype, np.number) for dtype in gene_expression_df.dtypes]]
     )
 
-    # exclude n_cells_tissue as we do not wish to roll it up
-    if "n_cells_tissue" in numeric_columns:
-        numeric_columns.remove("n_cells_tissue")
+    group_by_terms = list(universal_set_cell_counts_df.index.names)
+    available_combinations = set(universal_set_cell_counts_df.index.values)
 
-    # get the total number of cells per tissue to populate the n_cells_tissue in the added entries
-    n_cells_tissue_dict = gene_expression_df.groupby("tissue_ontology_term_id").first()["n_cells_tissue"].to_dict()
+    numeric_columns.remove("n_cells_tissue")
 
-    # get the set of available combinations of group-by terms from the aggregated cell counts
-    available_combinations = set(universal_set_cell_counts_df.index.values)
+    pivoted = gene_expression_df.pivot_table(
```

Comment:
pivot the tidy dataframe into a 2D array with indices corresponding to `group_by_terms` and columns corresponding to genes.